test/integration/reload_test.lua

```diff
@@ -59,7 +62,90 @@ local function reload_roles(srv)
     t.assert_equals({ok, err}, {true, nil})
 end
 
+local task_sleep_on_10_eval = [[
+    for i = 1,100 do
+        box.space.customers:insert({i})
+    end
+    local expirationd = require('expirationd')
+    local tuples_cnt = 0
+    local is_expired_sleep = function()
+        tuples_cnt = tuples_cnt + 1
+        if tuples_cnt == 10 then
+            require("fiber").sleep(60)
+        end
+        return true
+    end
+    task = expirationd.start("walk_all", box.space.customers.id, is_expired_sleep,
+                             {process_expired_tuple = function() return true end,
+                              force = true})
+
+    for i = 1,100 do
+        if tuples_cnt >= 10 then
+            break
+        end
+        require("fiber").yield()
+    end
+]]
+
+local task_first_tuple_eval = [[
+    local expirationd = require('expirationd')
+    local tuple = nil
+    local is_expired_tuple = function(arg, t)
+        if tuple == nil then
+            tuple = t
+        end
+        return true
+    end
+    task = expirationd.start("walk_all", box.space.customers.id, is_expired_tuple,
+                             {force = true})
+
+    for i = 1,100 do
+        if tuple ~= nil then
+            break
+        end
+        require("fiber").yield()
+    end
+
+    task:kill()
+
+    return tuple or {}
+]]
+
+function g.test_task_continue_after_reload(cg)
+    cg.cluster:server('s1-master').net_box:eval(task_sleep_on_10_eval)
+
+    reload_roles(cg.cluster:server('s1-master'))
+
+    local tuple = cg.cluster:server('s1-master').net_box:eval(task_first_tuple_eval)
+    t.assert_equals(tuple, {10})
```

Comment:
I don't get it. Shouldn't tuples from `{1}` to `{10}` expire on the first eval? (I don't have expirationd experience.)

Reply:
We set `process_expired_tuple`, so expirationd does not delete the tuples, just iterate.

Reply:
Because it does not actually iterate over {10} tuple, it sleep on it.

Reply:
Oh, yeah, I had reread the code and now I'm see it.